### PR TITLE
(DOCSP-28379) Reorganizes evergreen docs for Pine TOC

### PR DESCRIPTION
**Pull request by sarahsimpers**
### Description
Carries out the plan from [DOCSP-28379](https://jira.mongodb.org/browse/DOCSP-28379).

There is no net new content added here; anything that shows as new content is copied and pasted from existing pages that were deleted. The only change I made to content was changing the heading level if two pages were merged.

### Testing
I staged this locally in Pine and it rendered as expected, but will require changing the landing page in Pine for Evergreen to Home. I can make that change once this PR is merged.

### Documentation
N/A

### CODE REVIEW

**Comment by sarahsimpers** [comment]
@bsamek Can you please review this PR when you have time? Thank you!

File: docs/01-Home.md [hunk 1]
```diff
@@ -48,3 +48,13 @@ Once builds are scheduled, an internal heuristic decides how many cloud machines
 
 Evergreen sends an executable agent binary to each test machine, which begins running its given task.
 Logs are streamed back to the main server, along with system statistics and formatted test results.
+
+# Spruce: The New Evergreen UI
```
Comment (bsamek) [inline docs/01-Home.md:52]: should be second-level heading

```suggestion
## Spruce: The New Evergreen UI
```

File: docs/05-Reference/03-pull_request_template.md
Comment (bsamek) [inline docs/05-Reference/03-pull_request_template.md:1]: I moved this file out of `/docs` today, so it shouldn't be in the PR.

**Review by bsamek — APPROVED** [review]
lgtm with 2 nits

File: docs/05-Reference/03-pull_request_template.md
Comment (sarahsimpers) [reply docs/05-Reference/03-pull_request_template.md:1]: > @sarahsimpers I'm going to commit my suggestions and merge, and make the pine change, now.

Thank you!!

**Pull request merged**